dds/idl/dynamic_data_adapter_generator.cpp

```diff
@@ -400,12 +400,16 @@ namespace {
     }
 
     if (generate) {
+      NoSafetyProfileGuard nspg;
+      wrapper.generate_tag();
+
       DynamicDataAdapterGuard ddag;
 
-      be_global->impl_ <<
+      std::string impl_classname = "DynamicDataAdapterImpl<" + cpp_name + tag + " >";
+
+      be_global->header_ <<
         "template <>\n"
-        "class DynamicDataAdapterImpl<" << cpp_name << tag << " > : public DynamicDataAdapter_T<"
-          << cpp_name << "> {\n"
+        "class " << impl_classname << " : public DynamicDataAdapter_T<" << cpp_name << "> {\n"
```

Comment:
I explicitly avoided putting these in the headers because almost every single struct and union gets these and everything put in headers slows down builds.